app/controllers/posts_controller.rb

```diff
@@ -1,10 +1,51 @@
 class PostsController < ApplicationController
   def index
-    @posts = Post.all.order(user_id: :desc)
+    @posts = Post.includes([:user]).where(user_id: params[:user_id])
```

Comment:
**Highlights**
- Well done using the `includes` method for solving the n+1 query problems